test/TestLinearPremiumPriceOracle.js

```diff
@@ -0,0 +1,63 @@
+const ENS = artifacts.require('@ensdomains/ens/ENSRegistry');
+const BaseRegistrar = artifacts.require('./BaseRegistrarImplementation');
+const DummyOracle = artifacts.require('./DummyOracle');
+const LinearPremiumPriceOracle = artifacts.require('./LinearPremiumPriceOracle');
+
+const namehash = require('eth-ens-namehash');
+const sha3 = require('web3-utils').sha3;
+const toBN = require('web3-utils').toBN;
+
+contract('LinearPremiumPriceOracle', function (accounts) {
+    let priceOracle;
+
+    before(async () => {
+		ens = await ENS.new();
+		registrar = await BaseRegistrar.new(ens.address, namehash.hash('eth'));
+		await ens.setSubnodeOwner('0x0', sha3('eth'), registrar.address);
+		await registrar.addController(accounts[0]);
+
+        // Dummy oracle with 1 ETH == 2 USD
+        var dummyOracle = await DummyOracle.new(toBN(2000000000000000000));
+        // 4 attousd per second for 3 character names, 2 attousd per second for 4 character names,
+        // 1 attousd per second for longer names.
+        // Pricing premium starts out at 100 USD at expiry and decreases to 0 over 100k seconds (a bit over a day)
+        const premium = toBN("100000000000000000000");
+        const decreaseRate = toBN("1000000000000000");
+        priceOracle = await LinearPremiumPriceOracle.new(dummyOracle.address, [0, 0, 4, 2, 1], premium, decreaseRate);
+    });
+
+    it('should report the correct premium and decrease rate', async () => {
+        assert.equal((await priceOracle.initialPremium()).toString(), "100000000000000000000");
+        assert.equal((await priceOracle.premiumDecreaseRate()).toString(), "1000000000000000");
+    })
+
+    it('should return correct base prices', async () => {
+        assert.equal((await priceOracle.price("foo", 0, 3600)).toNumber(), 7200);
+        assert.equal((await priceOracle.price("quux", 0, 3600)).toNumber(), 3600);
+        assert.equal((await priceOracle.price("fubar", 0, 3600)).toNumber(), 1800);
+        assert.equal((await priceOracle.price("foobie", 0, 3600)).toNumber(), 1800);
+    });
+
+    it('should not specify a premium for first-time registrations', async () => {
```

Comment:
I don't fully understand this test. The premium is zero because duration is 0, not whether it's first-time registrations or not.  To test for people who are firs time registrations (or no more premium), I think you have to do something like this.

```
    it('should not have premium if decreasing period is over', async () => {
        const decreasingPeriod = 100000 // 100k sec
        const ts = (await web3.eth.getBlock('latest')).timestamp - (90 * 86400 + decreasingPeriod + 1);
        assert.equal((await priceOracle.premium("foobar", ts, 0)).toString(), 0);
        assert.equal((await priceOracle.price("foobar", ts, 3600)).toNumber(), 0);
    });
```

However, running the above test resulted in `revert`

Reply:
Premium doesn't depend on the duration, only on when the name was last registered.